includes/YesWiki.php

```diff
@@ -405,45 +405,48 @@ public function Redirect($url)
     }
 
     // returns just PageName[/method].
-    public function MiniHref($method = null, $tag = null)
+    public function MiniHref($method = null, $tag = null, $anchor = null)
     {
         if (! $tag = trim($tag)) {
             $tag = $this->tag;
         }
 
-        return $tag . ($method ? '/' . $method : '');
+        return $tag . ($anchor ? '#' . $anchor : '') . ($method ? '/' . $method : '');
     }
 
     // returns the full url to a page/method.
-    public function Href($method = null, $tag = null, $params = null, $htmlspchars = true)
+    public function Href($method = null, $tag = null, $params = null, $anchor = null, $htmlspchars = true)
     {
         if (! $tag = trim($tag)) {
             $tag = $this->tag;
         }
-        $href = $this->config["base_url"] . $this->MiniHref($method, $tag);
+        $href = $this->config["base_url"] . $this->MiniHref($method, $tag, $anchor);
         if ($params) {
+            $ampChar = $htmlspchars ? '&amp;' : '&';
             if (is_array($params)) {
                 $paramsArray = [];
                 foreach ($params as $key => $value) {
                     if ($value) {
                         $paramsArray[] = "$key=$value";
+                    } else {
+                        $paramsArray[] = "$key";
                     }
                 };
                 if (count($paramsArray)>0) {
-                    $params = implode("&", $paramsArray);
+                    $params = implode($ampChar, $paramsArray);
                 } else {
                     $params = '';
                 }
             }
-            $href .= ($this->config['rewrite_mode'] ? '?' : ($htmlspchars ? '&amp;' : '&')) . $params;
+            $href .= ($this->config['rewrite_mode'] ? '?' : $ampChar) . $params;
         }
         if (isset($_GET['lang']) && $_GET['lang']!='') {
             $href .= '&lang='.$GLOBALS['prefered_language'];
         }
         return $href;
     }
 
-    public function Link($tag, $method = null, $params = null, $text = null, $track = 1, $forcedLink = false)
+    public function Link($tag, $method = null, $params = null, $anchor = null, $text = null, $track = 1, $forcedLink = false)
```

Comment:
```suggestion
    public function Link($tag, $method = null, $params = null, $text = null, $track = 1, $forcedLink = false, $anchor = null)
```
c'était pas plus simple de faire cette modification et de ne pas toucher au code dans tous les autre fichiers (surtout que la fonction Link est un peu utilisée par tout le monde).
Par souci de rétrocompatibilité, je m'oppose à la modification de l'ordre des paramètre de Link sachant qu'il est assez facile de faire autrement.